misk/src/main/kotlin/misk/digester/Windower.kt

```diff
@@ -0,0 +1,108 @@
+package misk.digester
+import java.util.Calendar
+/** Window represents a time range */
+class Window (private val start: Calendar, private val end: Calendar) {
+  /** Returns true if the given time t falls within the window's Start <= t < End. */
+  fun contains(t: Calendar): Boolean {
+    return !t.before(start) && t.before(end)
+  }
+  /** Returns true when the other window has the same start and end times. */
+  fun equals(other: Window): Boolean {
+    return start.time == other.start.time && end.time == other.end.time
+  }
+  /** Returns string representing the start and end time */
+  fun string(): String {
+    return "[${start.time}, ${end.time})"
+  }
+}
+/** Windower contains multiple windows with a specified duration */
+class Windower {
+  private var windowSize: Int = 0
```

Comment:
Both `windowSize` and `startSecs` should be initialized in the constructor since they aren't known prior to that point.